core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -753,13 +750,17 @@ object Doc {
    *     doc.render(10) // produces "1, 2, 3"
    */
   def fill(sep: Doc, ds: Iterable[Doc]): Doc = {
+    def concat(x: Doc, y: Doc): Doc = x match {
```

Comment:
this is not stack safe: if we have a right associated `x` we will recurse the full depth of it into the stack.

Reply:
A stack-safe variant is easy enough, but I think you're right that we don't need to enforce this as strictly as this change does.